### ISSUE TITLE
Updating the organization account name triggers an error

### ISSUE THREAD

**Issue by s-mm** [pypi/warehouse#12418]
When the organization account name is updated, the update triggers an SQLAlchemy error. 

To recreate the error-

- Click on 'Your organizations'
- Click on 'Manage' for a specific organization
- Click on 'Settings'
- Scroll to the bottom of the page and click on 'Change organization account name'
- Enter the New organization account name and Current organization account name
- Click on 'Change organization account name'

If the new name that is chosen was used previously, it leads to the error shown below. Essentially, we cannot revert to a previous org account name.

<img width="1429" alt="org_name_error" src="https://user-images.githubusercontent.com/37237726/197764693-a2244868-5d46-4a82-9be9-354ae4754012.png">

**Comment by s-mm** [comment]
Can the name be changed to an account name that is not used currently? Should we limit the number of times an org account name can be changed?

**Comment by ewdurbin** [comment]
I'm able to reproduce this error in a couple of ways:

1) By changing only the capitalization of an organization, e.g. `psf` -> `PSF`
2) By trying to change the organization name back to a previous one, e.g. `psf` -> `pyfound` -> `psf`

For 1), I think this is a case where the normalization of the name for the catalog comes into play, so it's a matter of checking that before writing the catalog entry.

For 2), Since the catalog tracks what `organization_id` held a given name, we should allow reclaiming of previous names.

On a broader note, It may make sense for us to use the catalog as a means of determining how many normalized names a given organization has used and limit the number of renames allowed before needing to request support. Otherwise once an organization is approved, they could "hop scotch" through a bunch of names to block them from use them without admin intervention. This is probably best addressed in a separate issue.
